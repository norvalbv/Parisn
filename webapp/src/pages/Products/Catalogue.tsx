import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { PRODUCT_1_IMAGE } from '../../constants';
import io from 'socket.io-client';

const Catalogue = () => {
  let socket = io('ws://localhost:8000', {
    withCredentials: true,
  });

  const [viewCount, setViewCount] = useState({});
  const products = [
    {
      id: 1,
      name: 'Product 1',
      title: 'Product 1',
      image: PRODUCT_1_IMAGE,
      currentPrice: '1.00',
      viewers: socket.on('from', (amount) => {
        console.log(amount);
        setViewCount((prev) => {
          return {
            ...prev,
            views: amount.views,
            page: amount.page,
          };
        });
      }),
    },
    {
      id: 2,
      name: 'Product 2',
      title: 'Product 2',
      image: PRODUCT_1_IMAGE,
      currentPrice: '1.00',
      viewers: socket.on('from', (amount) => {
        setViewCount((prev) => {
          return {
            ...prev,
            views: amount.views,
            page: amount.page,
          };
        });
      }),
    },
    {
      id: 3,
      name: 'Product 3',
      title: 'Product 3',
      image: PRODUCT_1_IMAGE,
      currentPrice: '1.00',
      viewers: socket.on('from', (amount) => {
        setViewCount((prev) => {
          return {
            ...prev,
            views: amount.views,
            page: amount.page,
          };
        });
      }),
    },
  ];

  console.log(viewCount);

  return (
    <div className="grid grid-cols-3 pt-24 pb-8">
      {products.map((product, i) => (
        <div
          key={product.id}
          className="flex flex-col flex-wrap justify-center items-center border-r border-secondary-neutral last:border-none pt-10"
        >
          <p className="underline">{product.title}</p>
          <Link to="/shop-item" className="flex justify-center items-center">
            <img src={product.image} alt={product.name} className="w-3/4 cursor-pointer"></img>
          </Link>
          <p>Current Price: Â£{product.currentPrice}</p>
          {/* <p>Current Viewers: {Object.keys(viewCount[i]).length === 0 ? 0 : viewCount[i]}</p> */}
        </div>
      ))}
    </div>
  );
};

export default Catalogue;
