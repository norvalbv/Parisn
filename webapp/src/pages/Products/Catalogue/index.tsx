import React, { ReactElement, useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useProducts } from '../../../services/DataApiService';
import LiveViewers from '../../../components/LiveViewers';
import { MockData } from '../../../../../server';

const Catalogue = (): ReactElement => {
  const [products, setProducts] = useState<null | MockData[]>();
  useEffect(() => {
    (async () => {
      const { data } = await useProducts();
      setProducts(data);
    })();
  }, []);

  if (!products) return <></>;

  return (
    <div className="grid grid-cols-3 pt-24 pb-8">
      {products.map((product) => (
        <div
          key={product.id as unknown as string}
          className="flex flex-col flex-wrap justify-center items-center border-r border-secondary-neutral last:border-none pt-10"
        >
          <p className="underline">{product.title as unknown as string}</p>
          <Link
            to={`/shop-item?product=${product.id}`}
            className="flex justify-center items-center"
          >
            <img
              src={product.image as unknown as string}
              alt={product.name as unknown as string}
              className="w-[24rem] h-[34rem] cursor-pointer"
            />
          </Link>
          {/* <p>Current Price: Â£{product.price}</p> */}
          <LiveViewers params={product.id as unknown as string} />
        </div>
      ))}
    </div>
  );
};

export default Catalogue;
