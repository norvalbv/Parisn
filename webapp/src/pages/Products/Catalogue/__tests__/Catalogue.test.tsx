import renderer from 'react-test-renderer';

import Catalogue from '../';

describe('<Catalogue />', () => {
  test('Catalogue props with mandatory props', async () => {
    const tree = renderer.create(<Catalogue />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
