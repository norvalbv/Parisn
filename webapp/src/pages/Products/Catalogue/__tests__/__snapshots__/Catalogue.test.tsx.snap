// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Catalogue /> Catalogue props with mandatory props 1`] = `null`;
